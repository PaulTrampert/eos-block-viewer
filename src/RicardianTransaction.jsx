import React from 'react';
import PropTypes from 'prop-types';
import RicardianAction from './RicardianAction';
import Transaction from './propTypes/Transaction';

const RicardianTransaction = ({
  transaction,
  abis
}) => {
  return (
    <>
      {transaction.actions.map((a, idx) => <RicardianAction key={idx} transaction={transaction} actionIndex={idx} abi={abis[a.account].abi} />)}
    </>
  )
}

RicardianTransaction.propTypes = {
  transaction: Transaction,
  abis: PropTypes.object
}

export default RicardianTransaction;