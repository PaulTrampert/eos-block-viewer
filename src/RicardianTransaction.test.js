import React from 'react';
import {shallow} from 'enzyme';
import RicardianTransaction from './RicardianTransaction';

describe('RicardianTransaction', () => {
  let subject;
  let abis;
  let transaction;

  beforeEach(() => {
    abis = {
      a: {
        abi: {
          version: '0.2.0'
        }
      },
      b: {
        abi: {
          version: '0.1.0'
        }
      }
    }
    transaction = {
      actions: [
        {
          account: 'a'
        },
        {
          account: 'b'
        },
        {
          account: 'a'
        },
      ]
    }

    subject = shallow(<RicardianTransaction abis={abis} transaction={transaction} />);
  });

  it('renders an action for each action on the transaction', () => {
    expect(subject).toMatchSnapshot();
  });
});