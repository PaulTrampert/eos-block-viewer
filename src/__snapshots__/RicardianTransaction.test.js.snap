// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RicardianTransaction renders an action for each action on the transaction 1`] = `
<Fragment>
  <RicardianAction
    abi={
      Object {
        "version": "0.2.0",
      }
    }
    actionIndex={0}
    key="0"
    transaction={
      Object {
        "actions": Array [
          Object {
            "account": "a",
          },
          Object {
            "account": "b",
          },
          Object {
            "account": "a",
          },
        ],
      }
    }
  />
  <RicardianAction
    abi={
      Object {
        "version": "0.1.0",
      }
    }
    actionIndex={1}
    key="1"
    transaction={
      Object {
        "actions": Array [
          Object {
            "account": "a",
          },
          Object {
            "account": "b",
          },
          Object {
            "account": "a",
          },
        ],
      }
    }
  />
  <RicardianAction
    abi={
      Object {
        "version": "0.2.0",
      }
    }
    actionIndex={2}
    key="2"
    transaction={
      Object {
        "actions": Array [
          Object {
            "account": "a",
          },
          Object {
            "account": "b",
          },
          Object {
            "account": "a",
          },
        ],
      }
    }
  />
</Fragment>
`;
